crates/proof-of-sql/src/base/database/owned_table_utility.rs

```diff
@@ -231,3 +235,66 @@ pub fn timestamptz<S: Scalar>(
         OwnedColumn::TimestampTZ(time_unit, timezone, data.into_iter().collect()),
     )
 }
+
+/// Creates a (Identifier, OwnedColumn) pair for a timestamp column.
+/// This is primarily intended for use in conjunction with [owned_table].
+///
+/// # Parameters
+/// - `name`: The name of the column.
+/// - `time_unit`: The time unit of the timestamps.
+/// - `timezone`: The timezone for the timestamps.
+/// - `data`: The data for the column, provided as an iterator over `i64` values representing time since the unix epoch.
+///
+/// # Example
+/// ```
+/// use proof_of_sql::base::{database::owned_table_utility::*,
+///     scalar::Curve25519Scalar,
+/// };
+/// use proof_of_sql_parser::{
+///    posql_time::{PoSQLTimeZone, PoSQLTimeUnit}};
+///
+/// let result = owned_table::<Curve25519Scalar>([
+///         timestamptz("event_time",
+///         PoSQLTimeUnit::Second,
+///         vec![
+///             "1969-12-31T23:59:59Z", // One second before the Unix epoch
+///             "1970-01-01T00:00:00Z", // The Unix epoch
+///             "1970-01-01T00:00:01Z", // One second after the Unix epoch
+///         ].iter().map(|s| s.to_string())),
+/// ]);
+/// ```
+pub fn timestamptz<S: Scalar>(
+    name: impl Deref<Target = str>,
+    time_unit: PoSQLTimeUnit,
+    data: impl IntoIterator<Item = String>,
+) -> (Identifier, OwnedColumn<S>) {
+    let parsed_data: Vec<PoSQLTimestamp> = data
+        .into_iter()
+        .map(|timestamp_str| {
+            PoSQLTimestamp::try_from(&timestamp_str).expect("Failed to parse timestamp")
+        })
+        .collect();
+
+    // Build the OwnedColumn using the time unit from the first timestamp in parsed_data
+    let result = (
+        name.parse().unwrap(),
+        OwnedColumn::TimestampTZ(
+            parsed_data
+                .first()
+                .expect("No timestamps provided")
+                .timeunit,
```

Comment:
This should fix the tests:
```suggestion
            time_unit,
```